tensorflow/core/ops/math_ops_test.cc

```diff
@@ -604,4 +604,22 @@ TEST(MathOpsTest, SobolSample) {
 
   INFER_OK(op, "[];[];[]", "[?,?]");
 }
+
+TEST(MathOpsTest, EqualOp) {
+  ShapeInferenceTestOp op("Equal");
+  AddNodeAttr("incompatible_shape_error", true, &op.node_def);
```

Comment:
(Optional) This attribute should not be necessary - do tests fail without it?

Reply:
@mdanatg `incompatible_shape_error` Attr is part of the `Equal` op due to the following line, so it is needed.
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/ops/math_ops.cc#L718